### PR TITLE
Fix: #6357, Force Login rate element to be center of parent

### LINKED ISSUE #6357: "Log in to rate this …" offset in Firefox ESR 52.9.0 and Waterfox
For example: 

2018-09-22 01 57 25 terminals

– Waterfox 56.2.3 on FreeBSD-CURRENT, with a new profile. 

Affecting pages such as https://addons.mozilla.org/addon/loremizer/ 

Not reproducible in Firefox, Chromium, Surf or Web. 

Layout in browsers such as Falkon is quite different but (at a glance) not comparably bugged.

### PR DESCRIPTION
**Pull request by HosseinMarzban**
#6357: Force Login rate element to be center of parent.

### CODE REVIEW

**Comment by SeanPrashad** [comment]
Hey @HosseinMarzban! Before the AMO team gets here, we can polish up your PR a bit more 😀

As @klymenkoo is handling #6762, we can go ahead and revert that change as his pull request will fix it.

**Comment by HosseinMarzban** [comment]
@SeanPrashad hi, could you help me, `travic` gives me an error that mentions that before commit you must run `prettier-dev` so I did it, but it changes lost of files, now as you said i run `git reset --hard HEAD~1`, whats next step?

**Comment by SeanPrashad** [comment]
Hmm, running the command `yarn prettier-dev` should only format files that you have changed.

Anyhow, you need to force push next using `git push -f origin master`. Let me know if that doesn't work!

**Comment by HosseinMarzban** [comment]
that confuses me, I just change two `SCSS` files I don't know why `yarn prettier-dev` change that files!
now I run `git push -f origin master` and shows me `Everything up-to-date`, what's next?

**Comment by SeanPrashad** [comment]
We should not see the commit 9f7e78b if it was successfully deleted.

Can you run `git remote -v` and paste the results here?

**Comment by SeanPrashad** [comment]
I'm not sure if this is a possibility the devs are OK with but you could add me as a collaborator to your repo so we could take a look at getting it fixed together 👊🏽

**Comment by SeanPrashad** [comment]
Looking at it now!

**Comment by SeanPrashad** [comment]
@HosseinMarzban Can you run `git push -f` while on your master branch?

**Comment by HosseinMarzban** [comment]
> Looking at it now!

okay thanks, now I must just run `commit pull` and then commit again?
what was wrong, and where is the problem coming from?

**Comment by HosseinMarzban** [comment]
> git push -f

don

**Comment by HosseinMarzban** [comment]
@SeanPrashad test failed again, this time all test failed, last time just linter test failed! whats wrong? :confused:

**Comment by SeanPrashad** [comment]
@willdurand @kumar303 @bobsilverberg  Travis's `prettier-dev` check fails if a user creates a PR that is on the `master` branch. This isn't the proper workflow for Git but the [docs don't mention creating a new branch](https://github.com/HosseinMarzban/addons-frontend/blob/master/CONTRIBUTING.md) for your work.

**Comment by HosseinMarzban** [comment]
that's right Travis always right:), I fixed it that linter problem before, but what about `COMMAND=prettier-ci` it says run `yarn prettier-dev`  before commit but when its don, `prettier-dev` changed lots of files, is that okay?

**Comment by SeanPrashad** [comment]
`yarn prettier-dev` fixes linting and style issues by comparing your branch to the `master` branch. In this case, you didn't create a new branch for your work, so I think its trying to compare `master` to `master`.

Running `yarn prettier` should fix up all files.

**Comment by championshuttler** [comment]
Hi @HosseinMarzban , thanks for working so far could you please revert back the changes for the `yarn.lock` file , these are not related to the issue we are looking to solve in this PR.

File: src/ui/components/Rating/styles.scss [hunk 1]
```diff
@@ -69,6 +69,7 @@
   .Rating-star {
     background-image: url('./img/open-star-dim-yellow.svg');
     cursor: pointer;
+    outline: none;
```
Comment (championshuttler) [inline src/ui/components/Rating/styles.scss]: As @SeanPrashad suggested, @HosseinMarzban  we need to revert it as PR #6762 is looking to solve the issue #6755.

File: src/amo/components/RatingManager/styles.scss [hunk 2]
```diff
@@ -12,6 +12,9 @@
   padding: 2px 16px;
   position: absolute;
   width: 100%;
+  top:50%;
+  left: 50%;
+  transform: translate(-50%,-50%)
```
Comment (championshuttler) [inline src/amo/components/RatingManager/styles.scss]: Hi @HosseinMarzban can we convert this whole section to something like this as i am suggesting here. I think that will solve the `stylelint` and `prettier` issues.
```
.RatingManager-log-in-to-rate-button {
  font-size: $font-size-s;
  padding: 2px 16px;
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  width: 100%;
}
```

**Comment by HosseinMarzban** [comment]
tanks @championshuttler and @SeanPrashad for helping, really I have no idea why this happened and where exactly did I make a mistake!? if you see now the `Travice` test and problem, it looking for this file `bin/build-checks.js` and could not find it, but we have this file in project file tree!

File: src/amo/components/RatingManager/styles.scss (hunk 2 above)
Comment (SeanPrashad) [reply src/amo/components/RatingManager/styles.scss]: This is exactly what we need!

**Comment by SeanPrashad** [comment]
We're back to the problem of adding too many files @HosseinMarzban:

image

What are the commands you're running to get this to happen?

**Comment by HosseinMarzban** [comment]
> `bin/build-checks.js` was complaining about [`crlf` vs `lf`](https://help.github.com/articles/dealing-with-line-endings/) in multiple files. I've gone ahead and corrected it, and also resetted `yarn.lock` to what it should be.
> 
> Be very careful if you're adding files to be staged using `git add .` - this can sometimes add files that we shouldn't be touching like `yarn.lock`.
> 
> Let's see what Travis says now 🙏🏽

you right, my bad, that's one of the problems come from, eslint must fixe this line stuff and for that, I add a new script in 'pakage.json' to fix all line automatically base on eslint config.

**Comment by SeanPrashad** [comment]
No worries - this is a good learning experience! We are **super** close to getting this ready - Travis is complaining about two little things:

```
src/amo/components/RatingManager/styles.scss
 15:3  ✖  Expected top to come before       order/properties-alphabetical-order
          width                                                                
 16:3  ✖  Expected left to come before top  order/properties-alphabetical-order
```

Can you pull in my changes using `git pull --rebase` and fix them up? If you get stuck, we could chat over in #amo on Mozilla's IRC server. Details on how to connect can be found [here](https://wiki.mozilla.org/IRC).

**Comment by SeanPrashad** [comment]
Also let's update the title of this pull request to be `Force Login rate element to be center of parent` and also remove `#6762: @klymenkoo fix this issue by add outline:none in start rate element.` from your original message as that is handled by someone else's pull request now.

Usually a pull request fixes one issue at a time! In your case, this pull request should fix one thing, which is issue #6357

**Comment by willdurand** [comment]
Thanks for the PR! I don't know if this is the proper fix given that it mostly works everywhere else, but I will think about it!

**Comment by caitmuenster** [comment]
Thank you so much, @HosseinMarzban! Your contributions have been added to our [recognition wiki](https://wiki.mozilla.org/Add-ons/Contribute/Recognition). 🎉

Would you be interested in creating an account on [mozillians.org](https://mozillians.org/)? I'd be happy to vouch for you!

**Comment by HosseinMarzban** [comment]
> Thank you so much, @HosseinMarzban! Your contributions have been added to our [recognition wiki](https://wiki.mozilla.org/Add-ons/Contribute/Recognition). 🎉
> 
> Would you be interested in creating an account on [mozillians.org](https://mozillians.org/)? I'd be happy to vouch for you!

Tanks @caitmuenster :blush: , this is my mozillians acount link:

https://mozillians.org/en-US/u/HosseinMarzban/

**Comment by caitmuenster** [comment]
Awesome! You are vouched. 

Welcome onboard! I look forward to seeing you around. :)

**Pull request merged**